### DISCUSSION TITLE
[Feature Request] hope enabling disable vertical scroll and fill the document according to the content

### DISCUSSION THREAD

**Discussion by zhwaaaaaa** [microsoft/monaco-editor#3677] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

Now I require setting a height value for the container and initialize the editor. when the content of editor growing up, the scrollbar emerged.
I hope that Monaco Editor can fill document flow of browser according to the content without scrollbar, and also not need setting an initial height value for the container.
Thank you

**Comment by ricsam** [comment]
You can set the height of the editor to `code.split('\n').length * 18`, each line has the height 18. To remove the scrollbar set the editor option `scrollBeyondLastLine` to false:
```js
// set the initial options or update the options after mounting the editor:
monaco.editor.updateOptions({ scrollBeyondLastLine: false });
```

**Reply by xzifan** [reply]
This won't work well if you use `editor.changeViewZones` to add custom areas. In that case, all row height should be taken into calculation of editor height. 

In this Feature Request, all we need is to disable the default optimization of Virtual Scroll and render all lines and tokens at once.
